_drafts/github-actions-for-stm32cubeide.md

```diff
@@ -0,0 +1,349 @@
+---
+title: GitHub Actions for STM32CubeIDE
+description: Using GitHub Actions to automate building STM32CubeIDE projects.
+author: noah
+---
+
+<!-- excerpt start -->
+
+In this article, we will explore how to use GitHub Actions to automate building
+STM32CubeIDE projects. Eclipse-based IDEs like STM32CubeIDE are often used for
+developing embedded systems, but can be a little tricky to build in a headless
+environment.
+
+<!-- excerpt end -->
+
+{% include newsletter.html %}
+
+{% include toc.html %}
+
+
+## STM32CubeIDE, Docker, and GitHub Actions
+
+![Picture of GitHub Actions calling STM32CubeIDE in a Docker container](/img/stm32cubeide/arch.excalidraw.svg)
+
+The STM32 series of microcontrollers from STMicroelectronics are very popular in
+the embedded industry. ST provides an in-house Eclipse-based IDE, [STM32CubeIDE](https://www.st.com/en/development-tools/stm32cubeide.html),
+for developing projects using these chips.
+
+The STM32CubeIDE provides an all-in-one starting point for STM32-based projects,
+with example code, drivers, and middleware, and tools for configuring
+peripherals and generating code.
+
+This IDE comes with both the graphical frontend (using Eclipse) and the
+necessary tools for building, flashing, and debugging STM32 projects (it bundles
+a pretty standard GCC ARM toolchain and the ST-Link tools for flashing and
+debugging).
+
+Typically developers install the STM32CubeIDE on their local machines and build
+projects using the GUI, but that manual and UI-based workflow doesn't fit well
+with modern CI systems- we want to build our projects automatically, without
+clicking through a UI, and in a reproducible way.
+
+To accomplish this, we'll use Docker to create a containerized environment with
+STM32CubeIDE and its dependencies, and GitHub Actions to automate the build
+process!
+
+## Step 1: Setting Up the Docker Image
+
+> Note: STM32CubeIDE generates `Makefile`'s when building a project- these
```

Comment:
Nice callout here. I'd imagine someone would ask about this, so it is good to get ahead of it.